### PR TITLE
Add the swift compiler version to the matrix in the SwiftFormat yaml.

### PR DESCRIPTION
**Pull request by hjyamauchi**

### CODE REVIEW

**Review by compnerd — COMMENTED** [review]
Thanks!  This looks like a step in the right direction.  I wonder if we should make the matrix the compiler versions and make the version of SwiftFormat an input.

File: .github/workflows/SwiftFormat.yml [hunk 1]
```diff
@@ -56,8 +58,8 @@ jobs:
         with:
           draft: true
           prerelease: true
-          release_name: SwiftFormat-${{ matrix.tag }}
-          tag_name: SwiftFormat-${{ matrix.tag }}
+          release_name: SwiftFormat-${{ matrix.tag }}-${{ matrix.swift_tag }}
```
Comment (mangini) [inline .github/workflows/SwiftFormat.yml:61]: The release name will be:
`SwiftFormat-0.51.10-DEVELOPMENT-SNAPSHOT-2023-05-22-a`

Which can be a bit confusing to parse. What about:
`SwiftFormat-${{ matrix.tag }}_for_swift_toolchain_${{ matrix.swift_tag }}`

Which would become, for example:
`SwiftFormat-0.51.10_for_swift_toolchain_DEVELOPMENT-SNAPSHOT-2023-05-22-a`

Comment (compnerd) [reply .github/workflows/SwiftFormat.yml:61]: Can we do `SwiftFormat-0.51.10 (Swift 5.8)` or `SwiftFormat-0.51.10 (Swift development)`?

**Comment by mangini** [comment]
> Thanks! This looks like a step in the right direction. I wonder if we should make the matrix the compiler versions and make the version of SwiftFormat an input.

This is actually what we have discussed initially. I like it.

File: .github/workflows/SwiftFormat.yml (hunk 1 above)
Comment (compnerd) [inline .github/workflows/SwiftFormat.yml:61]:
```suggestion
          release_name: SwiftFormat-${{ matrix.tag }} (${{ matrix.swift_branch }})
```

Comment (mangini) [reply .github/workflows/SwiftFormat.yml:61]: > Can we do `SwiftFormat-0.51.10 (Swift 5.8)` or `SwiftFormat-0.51.10 (Swift development)`?

Yes! I think the release name can have parenthesis and spaces, if so, it is much better :-)

**Review by compnerd — APPROVED** [review]

**Pull request merged**